src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -205,6 +205,103 @@ class Target
             await VerifyAnalyzerAsyncCs(source);
         }
 
+        [Fact]
+        public async Task SupportedOnOsx_GuardedWithIsMacOS()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+class Test
+{
+    public void Api_Usage()
+    {
+        if (OperatingSystem.IsWindows() ||
+            OperatingSystem.IsLinux() ||
+            OperatingSystem.IsMacOS())
+        {
+            Api();
+            Api2();
+        }
+
+        [|Api()|]; // This call site is reachable on all platforms. 'Test.Api()' is only supported on: 'MacOS/OSX', 'Linux', 'windows'.
+        [|Api2()|]; // This call site is reachable on all platforms. 'Test.Api2()' is only supported on: 'MacOS/OSX', 'Linux', 'windows'.
+    }
+
+    [SupportedOSPlatform(""macos"")]
+    [SupportedOSPlatform(""windows"")]
+    [SupportedOSPlatform(""Linux"")]
+    void Api() { }
+
+    [SupportedOSPlatform(""windows"")]
+    [SupportedOSPlatform(""Osx"")]
+    [SupportedOSPlatform(""Linux"")]
+    void Api2() { }
+}";
+
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task UnsupportedOnOsx_GuardedWithIsMacOS()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+class Test
+{
+    public void Api_Usage()
+    {
+        if (!OperatingSystem.IsWindows() &&
+            !OperatingSystem.IsLinux() &&
+            !OperatingSystem.IsMacOS())
+        {
+            Api();
+        }
+
+        [|Api()|]; // This call site is reachable on all platforms. 'Test.Api()' is unsupported on: 'MacOS/OSX', 'windows'.
```

Comment:
Do we want at least one test that verifies the diagnostic message/argument for these cases? I see that all the added test have annnotation based verification, and will pass even if diagnostic message does not have the expected terminology.

Reply:
For this test better not add message/argument verification because for multiple platforms message the order of the platforms appears in the list not consistent. But i will add such verification for tests flag only on platform